types/express-serve-static-core/.eslintrc.json

```diff
@@ -2,6 +2,7 @@
     "rules": {
         "@definitelytyped/no-type-only-packages": "off",
         "@typescript-eslint/naming-convention": "off",
-        "@typescript-eslint/no-empty-interface": "off"
+        "@typescript-eslint/no-empty-interface": "off",
+        "@typescript-eslint/no-misused-promises": "error"
```

Comment:
This is the linting issue that's giving troubles in #50871 and the reason for this PR. Using typescript-eslint's `strict-type-checked` is a good way to ensure code quality/safety/maintainability and this rule is part of it.